compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIELogicalObjFifoSplittingUtils.cpp

```diff
@@ -305,17 +296,50 @@ static LogicalResult checkWhetherSplitIsPossible(
   return success();
 }
 
-// Given a vector of L2->L1 Dma ops' perform the splitting :-
-// 1. Check if the splitting can be performed or not. If not possible, bail out.
-// 2. For the split dimension inferred set offset = 0 and size as 1 for L2 and
-//    L3.
-// 3. Now traverse each L2->L1 Dma op and perform the following :-
-//    a) Create a new L2 AllocOp based on the updated size (step 3 above) and
-//       create a logicalobjectfifo using the same.
-//    b) Split L3->L2 Dma op.
-//    c) SPlit L2->L1 Dma op.
-// 4. Delete old L2->L1, L3->L2 and corresponding AllocOps.
-LogicalResult splitLogicalObjectFifos(
+/// Utility to check if the L3->L2 dma is transposed on the L2 side.
+/// The intuition is if the sizes of L2 and L3 side are the same and L2 dma
+/// strides are in non-decreasing order from right to left, then the dma has
+/// transposition on L3 side, otherwise the transposition is on L2 side.
+static FailureOr<bool> isL2DmaTransposed(AMDAIE::DmaCpyNdOp l3ToL2DmaOp,
```

Comment:
Just realised, we should actually also ignore dimensions of size 1 and 0 when checking for decreasing dimensions, because there stride values mean nothing. Maybe that is why the shape comparison is needed

Reply:
Ok, I've updated with the 0/1 check on dimsize and it seems to work without the rank-shape check, which is hopefully an improvement and more reusable whenever we want to check for transposition in dma

Reply:
Vivian, hopefully this time you can just merge this PR into your branch (if you think it's a useful contribution)

Reply:
@newling Thanks for your contribution! What I don't understand is why the stride is non-static, then this might be a transpose. And if we use word "maybe" which means we are not sure if it's transposed.

Reply:
My thinking is, if the stride is `?` it could be a very large or a very small stride, which means this might be transpose. i.e. it is "maybe" a transpose. 

Example 

sizes = [4,4,4] 
strides = [16, ?, 1]

If the `?` is greater than 16, then this represents a transpose. If it <= 16, then it does not represent a transpose. 

I guess we could have had signature `isDefinitelyTranspose` and returned false for `?` (because with `?` it isn't _definitely_ a transpose)

It's all very edge-casey though as I don't think we'll ever have dynamic strides.

Reply:
Yeah, I thought that's what you had. I just think returning failure increases the complexity _slightly_, but I don't have a strong preference either way if it's documents what the true/failure/false cases denote.